crates/compiler/builtins/roc/List.roc

```diff
@@ -1383,3 +1385,44 @@ concatUtf8 : List U8, Str -> List U8
 
 expect (List.concatUtf8 [1, 2, 3, 4] "🐦") == [1, 2, 3, 4, 240, 159, 144, 166]
 
+## Run an effectful function for each element on the list.
+##
+## ```roc
+## List.forEach! ["Alice", "Bob", "Charlie"] \name ->
+##     createAccount! name
+##     log! "Account created"
+## ```
+##
+## If the function might fail or you need to return early, use [forEachTry!].
+forEach! : List a, (a => {}) => {}
+forEach! = \list, func! ->
+    when list is
+        [] ->
+            {}
+
+        [elem, .. as rest] ->
+            func! elem
+            forEach! rest func!
+
```

Comment:
Would these be faster if I didn't use pattern matching and wrote them like how we do `walk`? https://github.com/roc-lang/roc/blob/b7c2cb084e15a56320e652ad3c77fdb8262bc1a4/crates/compiler/builtins/roc/List.roc#L499-L509

I wonder if that used to be faster but no longer is because of seamless slices or something.